src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -228,6 +241,27 @@ app.directive('dashboardApp', function ($injector) {
         $scope.indexPatterns = dashboardState.getPanelIndexPatterns();
       };
 
+      $scope.filter = function (field, value, operator, index) {
+        queryManager.add(field, value, operator, index);
+        updateState();
+      };
+
+
+      $scope.$watch('model.query', (newQuery) => {
+        $scope.model.query = migrateLegacyQuery(newQuery);
+        dashboardState.applyFilters($scope.model.query, filterBar.getFilters());
+        $scope.refresh();
+      });
+
+      $scope.$watch(() => dashboardState.getAppState().$newFilters, function (filters = []) {
```

Comment:
I moved all the filter handling logic into the `query_manager` module, but there's still a bit of tech debt here with this watch. I've got another branch where I'm working on removing the need for a watch, but I don't think it's necessary for this PR.